eng/pipelines/pr.yml

```diff
@@ -4,57 +4,73 @@ pr:
   - main
   - release/*
 
-variables:
-  - template: templates/variables/globals.yml
+extends:
+  template: /eng/pipelines/templates/1es-redirect.yml
+  parameters:
+    variables:
+      - template: /eng/pipelines/templates/variables/globals.yml@self
+    stages:
+      - stage: Build
+        jobs:
+          - job: Build_And_Test_Windows
+            displayName: Build And Test Windows
 
-jobs:
-  - job: Build_And_Test
-    displayName: Build And Test
+            variables:
+              TYPESPEC_VS_CI_BUILD: true # Enable official Visual Studio extension build
+              TYPESPEC_SKIP_DOCUSAURUS_BUILD: true # Disable docusaurus build
 
-    variables:
-      TYPESPEC_VS_CI_BUILD: true # Enable official Visual Studio extension build
-      TYPESPEC_SKIP_DOCUSAURUS_BUILD: true # Disable docusaurus build
-
-    strategy:
-      matrix:
-        # Node 18
-        "Linux - Node 18.x":
-          pool: azsdk-pool-mms-ubuntu-2004-general
-          imageName: ubuntu-20.04
-          nodeVersion: 18.x
-
-        "Windows - Node 18.x":
-          pool: azsdk-pool-mms-win-2022-general
-          imageName: windows-2022
-          nodeVersion: 18.x
-
-        # Node 20
-        "Linux - Node 20.x":
-          pool: azsdk-pool-mms-ubuntu-2004-general
-          imageName: ubuntu-20.04
-          nodeVersion: 20.x
-
-        "Windows - Node 20.x":
-          pool: azsdk-pool-mms-win-2022-general
-          imageName: windows-2022
-          nodeVersion: 20.x
-
-    pool:
-      name: $(pool)
-      vmImage: $(imageName)
-
-    steps:
-      - template: ./jobs/build-and-test.yml
-        parameters:
-          nodeVersion: $(nodeVersion)
-          poolName: $(pool)
-
-  - template: ./jobs/publish-artifacts.yml
-
-  - job: docker_build
-    pool:
-      name: azsdk-pool-mms-ubuntu-2004-general
-      vmImage: ubuntu-20.04
-    steps:
-      - script: docker build -f ./docker/Dockerfile .
-        displayName: Docker build
+            strategy:
+              matrix:
+                "Node 18.x":
```

Comment:
This had to be split in 2 separate matrix?

Reply:
The `os:` parameter in `pool` has to be a compile time value, so we can't use a matrix value there